code/game/gamemodes/devil/devil_game_mode.dm

```diff
@@ -0,0 +1,65 @@
+/datum/game_mode/devil
+	name = "devil"
+	config_tag = "devil"
+	protected_jobs = list("Lawyer", "Librarian", "Chaplain", "Head of Security", "Captain", "AI")
+	required_players = 2
+	required_enemies = 1
+	recommended_enemies = 4
+
+	var/traitors_possible = 4 //hard limit on devils if scaling is turned off
+	var/num_modifier = 0 // Used for gamemodes, that are a child of traitor, that need more than the usual.
+	var/objective_count = 2
+	var/minimum_devils = 1
+	var/devil_scale_coefficient = 6
+
+/datum/game_mode/devil/announce()
+	to_chat(world, {"<B>The current game mode is - Devil!</B><br>)
+		<span class='danger'>Devils</span>: Purchase souls and tempt the crew to sin!<br>
+		<span class='notice'>Crew</span>: Resist the lure of sin and remain pure!"})
+
+/datum/game_mode/devil/pre_setup()
+
+	if(config.protect_roles_from_antagonist)
+		restricted_jobs += protected_jobs
+
+	var/list/possible_devils = get_players_for_role(ROLE_DEVIL)
+	var/num_devils = 0
+	if(!possible_devils.len)
+		return 0
+	if(config.traitor_scaling)
+		num_devils = max(1, round((num_players())/(devil_scale_coefficient))+1)
+	else
+		num_devils = max(1, min(num_players(), traitors_possible))
+
+
+	for(var/j = 0, j < num_devils, j++)
+		if (!possible_devils.len)
+			break
+		var/datum/mind/devil = pick(possible_devils)
+		devils += devil
+		devil.special_role = ROLE_DEVIL
+		devil.restricted_roles = restricted_jobs
+
+		log_game("[devil.key] (ckey) has been selected as a [config_tag]")
+		possible_devils.Remove(devil)
+
+	if(devils.len < required_enemies)
+		return 0
+	return 1
+
+
+/datum/game_mode/devil/post_setup()
```

Comment:
You may need to update the antagHUD icons here (if you don't do it elsewhere in the code). Example: https://github.com/ParadiseSS13/Paradise/blob/master/code/game/gamemodes/cult/cult.dm#L182.